quark/plugin.py

```
@@ -501,7 +501,46 @@ def update_port(self, context, id, port):
             as listed in the RESOURCE_ATTRIBUTE_MAP object in
             quantum/api/v2/attributes.py.
         """
-        raise NotImplementedError()
+        LOG.info("update_port %s for tenant %s" % (id, context.tenant_id))
+        port_db = db_api.port_find(context, id=id, scope=db_api.ONE)
+        if not port_db:
+            raise exceptions.PortNotFound(port_id=id, net_id="")
+
+        port = port["port"]
+        if "fixed_ips" in port and port["fixed_ips"]:
```

Actually we agreed to implement it correctly and then add an extension that can optionally disable it if it displeases the implementer to have it. As for the actual call, all I'm saying is if it gets down to quark without any kind of body, I think we should raise an exception versus doing nothing